classes/controller/FrontController.php

```diff
@@ -1298,15 +1298,31 @@ protected function getThemeDir()
      */
     public function getLayout()
     {
+        return $this->context->shop->theme->getLayoutPath($this->getLayoutName());
+    }
+
+    /**
+     * Returns layout name for the current controller.
+     * Modules can override the layout if needed.
+     *
+     * @return string layout name
+     */
+    public function getLayoutName()
```

Comment:
```suggestion
    protected function getLayoutName()
```

I do"nt think this needs to be public since it's used internally only, but it can be protected to allow overriding it